frontend/public/components/masthead-toolbar.jsx

```diff
@@ -91,36 +91,41 @@ class MastheadToolbarContents_ extends React.Component {
     }
     // Ignore inactivity-timeout if set to less then 300 seconds
     if (window.SERVER_FLAGS.inactivityTimeout >= 300) {
-      window.addEventListener('click', _.throttle(this._onInactivityTimeout, 500));
-      window.addEventListener('keydown', _.throttle(this._onInactivityTimeout, 500));
+      window.addEventListener('click', _.throttle(this._resetInactivityTimeout, 500));
+      window.addEventListener('keydown', _.throttle(this._resetInactivityTimeout, 500));
     }
     this._updateUser();
   }
 
   componentDidUpdate(prevProps) {
+    const { flags, user } = this.props;
     if (
-      this.props.flags[FLAGS.OPENSHIFT] !== prevProps.flags[FLAGS.OPENSHIFT] ||
-      !_.isEqual(this.props.user, prevProps.user)
+      flags[FLAGS.OPENSHIFT] !== prevProps.flags[FLAGS.OPENSHIFT] ||
+      !_.isEqual(user, prevProps.user)
     ) {
       this._updateUser();
     }
+    // Start the inactivity countdown when the flag and user are loaded and the timeout is not set
+    if (
+      window.SERVER_FLAGS.inactivityTimeout >= 300 &&
+      !this.userInactivityTimeout &&
+      flags[FLAGS.OPENSHIFT] &&
```

Comment:
Are we meaning to disable this when not OpenShift or are we simply checking if the flag is set? If the latter, you should use `flagPending`. Otherwise, we should move the OpenShift-specific logic below for logout. I don't think we need to restrict this to OpenShift, though.

Reply:
I'm not sure we would need to block on the flag either. Presumably the flag will resolve well before the inactivity timeout kicks in.